numba_extensions/tests/test_torch_bf16.py

```diff
@@ -0,0 +1,81 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+import numba.cuda as cuda
+
+
+if cuda.get_current_device().compute_capability < (8, 0):
+    pytest.skip(
+        reason="bfloat16 require compute capability 8.0+",
+        allow_module_level=True,
+    )
+
+
+from bf16 import nv_bfloat16, get_shims
+import numpy as np
+
+
+# torch container should include ml-dtypes, etc.
+torch = pytest.importorskip("torch")
+
+# ml types should be installed in the torch container
+# ml_dtypes needed to patch np.dtype with bfloat16
+from ml_dtypes import bfloat16  # noqa: F401 E402
+from numba.np import numpy_support  # noqa: E402
+
+# what is the constructor vs what is the numba type ?
+numpy_support.FROM_DTYPE[np.dtype("bfloat16")] = nv_bfloat16.nb_type
```

Comment:
Now there's `_nbtype` attribute builtin to all Numbast bindings, so we don't need to add the `nb_type` attribute here.